rep/event.go

```diff
@@ -21,17 +31,22 @@ const (
 )
 
 type RetrievalEvent struct {
-	Code   RetrievalEventCode
+	Phase Phase
+	Code  RetrievalEventCode
+	// Status will include error messages for RetrievalEventFailure, other events use it to expand the Code description
 	Status string
+	// QueryResponse will be included for a RetrievalEventQueryAsk event
+	QueryResponse *retrievalmarket.QueryResponse
```

Comment:
No, because in autoretrieve we've switched from relying on those return values to only using the events for this data. Because we get them out of order (particularly the queries which happen in parallel), matching them up is a massive pain because of the asynchronicity combined with the parallel queries followed by sequential retrievals. The choice comes down to:

* Match up state between query return result and query events (what I was attempting to do previously); or
* Rely only on events and manage state locally within the event handling system (which requires more info from the events - both the `QueryResponse` and `RetrievalSize` in the state are for this).

The latter is working quite well, it just means these events need to be more informative.

One alternative to make it nicer code would be to make a `RetrievalEvent` interface and cast it to the specific form in the successful-query, successful-retrieval and failure cases where we have this additional data to fetch out of it. But that feels a little heavyweight to me.

The other alternative is to put this property onto the `RetrievalState` object. But I'm unsure about the event+state differentiation anyway, so :shrug:.